### PR TITLE
ci: fix test and prod deployments and add test files

### PR DESCRIPTION
**Pull request by RMCampos**
# Description

Create deployment files and strategy for test and prod.

Fixes [FSADT2-426](https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-426)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

This change will be tested when this PR gets merged. 

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

---
Thanks for the PR!
Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-backend-36-backend.apps.silver.devops.gov.bc.ca/) available
Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar-backend/actions/workflows/merge-main.yml)

### CODE REVIEW

File: .github/workflows/merge-main.yml [hunk 1]
```diff
@@ -205,54 +205,34 @@ jobs:
...
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
     environment:
       name: test
     env:
       ZONE: test
       NR_SPAR_BACKEND_VERSION: ${{ needs.release.outputs.version }}
+    strategy:
+      matrix:
+        name: [ backend ]
```
Comment (DerekRoberts) [inline .github/workflows/merge-main.yml:216]: For TEST and PROD include the init and database deployment, using `overwrite: false` to make sure they don't replace exciting deployments.

Here's an example from the QuickStart:

```
 deploys:
    name: Deploys
    needs:
      - builds
    runs-on: ubuntu-22.04
    strategy:
      matrix:
        name: [ init, database, backend ]
        include:
          - name: init
            file: common/openshift.init.yml
            overwrite: false
          - name: database
            file: database/openshift.deploy.yml
            overwrite: false
          - name: backend
            file: backend/openshift.deploy.yml
            overwrite: true
            parameters: -p MIN_REPLICAS=1 -p MAX_REPLICAS=2 -p BUILD=snapshot-${{ github.event.number }}
    steps:
      - uses: bcgov-nr/action-deployer-openshift@v1.0.1
        with:
          file: ${{ matrix.file }}
          oc_namespace: ${{ secrets.OC_NAMESPACE }}
          oc_server: ${{ secrets.OC_SERVER }}
          oc_token: ${{ secrets.OC_TOKEN }}
          overwrite: ${{ matrix.overwrite }}
          penetration_test: false
          verification_path: "actuator/health"
          parameters:
            -p ZONE=${{ github.event.number }} -p NAME=${{ github.event.repository.name }}
            -p PROMOTE=${{ github.repository }}/${{ matrix.name }}:${{ github.event.number }}
            ${{ matrix.parameters }}
```

File: .github/workflows/merge-main.yml [hunk 2]
```diff
@@ -366,79 +342,43 @@ jobs:
...
+    runs-on: ubuntu-22.04
     environment:
       name: prod
     env:
       ZONE: prod
       PREV: test
       NR_SPAR_BACKEND_VERSION: ${{ needs.release.outputs.version }}
+    strategy:
+      matrix:
+        name: [ backend ]
```
Comment (DerekRoberts) [inline .github/workflows/merge-main.yml:357]: Same as the last comment.  Include init and db.

File: test_deployment/openshift.deploy.yml [hunk 3]
```diff
@@ -0,0 +1,175 @@
+apiVersion: template.openshift.io/v1
```
Comment (DerekRoberts) [inline test_deployment/openshift.deploy.yml:1]: Please use the same templates across DEV, TEST and PROD.  Use parameters to handle different variables.

File: test_deployment/openshift.init.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
+apiVersion: template.openshift.io/v1
```
Comment (DerekRoberts) [inline test_deployment/openshift.init.yml:1]: Please use the same templates across DEV, TEST and PROD.  Use parameters to handle different variables.

**Review by DerekRoberts — CHANGES_REQUESTED** [review]

File: .github/workflows/merge-main.yml [hunk 5]
```diff
@@ -201,58 +180,42 @@ jobs:
...
+          - name: init
+            file: common/openshift.init.yml
+            overwrite: false
+          - name: database
+            file: database/openshift.deploy.yml
+            overwrite: false
+          - name: backend
+            file: backend/openshift.deploy.yml
+            overwrite: true
+            parameters: -p MIN_REPLICAS=1 -p MAX_REPLICAS=2 -p BUILD=test-${{ env.NR_SPAR_BACKEND_VERSION }}
```
Comment (DerekRoberts) [inline .github/workflows/merge-main.yml:204]: Remove MIN_REPLICAS and MAX_REPLICAS to use the template defaults, which are 3 and 5.

File: .github/workflows/merge-main.yml [hunk 6]
```diff
@@ -291,154 +254,49 @@ jobs:
...
```
Comment (DerekRoberts) [inline .github/workflows/merge-main.yml:280]: Remove MIN_REPLICAS and MAX_REPLICAS to use the template defaults, which are 3 and 5.

**Review by DerekRoberts — CHANGES_REQUESTED** [review]

File: .github/workflows/merge-main.yml (hunk 6 above)
Comment (RMCampos) [reply .github/workflows/merge-main.yml]: Done!

**Review by DerekRoberts — APPROVED** [review]
Genius!  Merge away.  ;)

**Pull request merged**